packages/angular/src/data-connect/index.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  type CreateMutationOptions,
+  type CreateMutationResult,
+  type CreateQueryOptions,
+  type CreateQueryResult,
+  injectMutation,
+  injectQuery,
+  QueryClient,
+  type QueryKey,
+} from "@tanstack/angular-query-experimental";
+
+import type { FirebaseError } from "firebase/app";
+
+import {
+  EnvironmentInjector,
+  inject,
+  type Injector,
+  signal,
+} from "@angular/core";
+import {
+  type CallerSdkType,
+  CallerSdkTypeEnum,
+  DataConnect,
+  executeMutation,
+  executeQuery,
+  type MutationRef,
+  type MutationResult,
+  type QueryRef,
+  type QueryResult,
+} from "@angular/fire/data-connect";
+function getQueryKey(queryRef: QueryRef<unknown, unknown>) {
+  const key: (string | Record<string, any>)[] = [queryRef.name];
+  if ("variables" in queryRef && queryRef.variables !== undefined) {
+    key.push(queryRef.variables as unknown as Record<string, any>);
+  }
+  return key;
+}
+export type FlattenedQueryResult<Data, Variables> = Omit<
+  QueryResult<Data, Variables>,
+  "data" | "toJSON"
+> &
+  Data;
+export interface CreateDataConnectQueryOptions<Data, Variables>
+  extends Omit<
+    CreateQueryOptions<
+      FlattenedQueryResult<Data, Variables>,
+      FirebaseError,
+      FlattenedQueryResult<Data, Variables>,
+      QueryKey
+    >,
+    "queryFn" | "queryKey"
+  > {
+  queryFn: () => QueryRef<Data, Variables>;
+}
+
+/**
+ * injectDataConnectQuery takes a query ref and returns a wrapper function around Tanstack's `injectQuery`
+ * @param queryRefOrOptionsFn Query Ref or callback function for calling a new query
+ * @returns {CreateQueryResult<FlattenedQueryResult<Data, Variables>>}
+ */
+export function injectDataConnectQuery<Data, Variables>(
+  queryRefOrOptionsFn:
+    | QueryRef<Data, Variables>
+    | (() => CreateDataConnectQueryOptions<Data, Variables>),
+  injector?: Injector,
+  _callerSdkType: CallerSdkType = CallerSdkTypeEnum.TanstackAngularCore
+): CreateQueryResult<FlattenedQueryResult<Data, Variables>, FirebaseError> {
+  const finalInjector = injector || inject(EnvironmentInjector);
+  const queryKey = signal<QueryKey>([]);
+
+  function fdcOptionsFn() {
+    const passedInOptions =
+      typeof queryRefOrOptionsFn === "function"
+        ? queryRefOrOptionsFn()
+        : undefined;
+
+    const modifiedFn = async (): Promise<
+      FlattenedQueryResult<Data, Variables>
+    > => {
+      const ref: QueryRef<Data, Variables> =
+        passedInOptions?.queryFn() ||
+        (queryRefOrOptionsFn as QueryRef<Data, Variables>);
+      // @ts-expect-error function is hidden under `DataConnect`.
+      ref.dataConnect._setCallerSdkType(_callerSdkType);
+      queryKey.set([ref.name, ref.variables]);
+      const { data, ...rest } = await executeQuery(ref);
+      return {
+        ...data,
```

Comment:
Why are we merging this? If someone has a query with a top-level field called `source`, `fetchTime` or `ref`, that field will be shadowed by the `...rest` below and they will not be able to access that field. The typing  `FlattenedQueryResult` above doesn't seem to reflect this behavior accurately either.

Furthermore, based on [this comment](https://github.com/firebase/firebase-js-sdk/blob/feb2c9dfa29c9dff01c1272e56f6258176dc6b3a/packages/data-connect/src/api/Reference.ts#L42) we may add future metadata fields there. If/when we do so, every new field we add will be a breaking change since it may shadow yet another top-level field.